### ISSUE TITLE
AnyConnect iOS cannot connect to VPN

### ISSUE THREAD

**Issue by GauthierPLM** [StreisandEffect/streisand#1396]
### Expected behavior:

After the creation of a new server and the configuration of AnyConnect, the app should be able to connect to the VPN.
The app is supposed to ask an username and a password for first connection.

### Actual Behavior:

When following the instructions to setup AnyConnect on iOS, everything goes well until connection which is always a failure when connecting.

No credentials were required when setting up the certificate nor for the first connection that failed.

### Steps to Reproduce:
1. Install a fresh Streisand server on Google Cloud.
2. Follow the instructions to setup AnyConnect
3. Try to connect. A message saying that an error occurred while trying to connect appears.

I attached a domain name to the server and Streisand configured a Let's Encrypt certificate. Connection to the server works well.
This also happens on an instance I setup in February which I didn't touched since.

The version of the AnyConnect app is 4.6.01099

**Comment by MRizkBV** [comment]
Just moved to Google Cloud now and I too am facing the same issue with iOS. (I do not use a domain, or Let's Encrypt and never did on DigitalOcean too but it worked fine there).

**Comment by alimakki** [comment]
Able to reproduce, will investigate.

**Comment by alimakki** [comment]
I'm guessing something has changed in the last few AnyConnect client updates and how certificates are validated as the app no longer seems to like it. On the other hand, username+password authentication *does* still work, so that may still be used as a workaround until more debugging is done.

**Comment by MRizkBV** [comment]
Unfortunately the legacy app is not compatible with iOS 12 Beta :/

**Comment by MRizkBV** [comment]
@alimakki Is there a simple way to enable username+password auth only? I am not an expert with OpenConnect. I only know how to control OpenVPN. Thanks.

**Comment by alimakki** [comment]
@MRizkBV The OpenConnect role already generates a username and password for authentication, check the Windows instructions. You'll just have to type in the IP address of the server.

**Comment by MRizkBV** [comment]
@alimakki Tried it but apparently iOS does not store the password which kills the purpose of the On Demand connect feature. This is why I was using OpenConnect in the first place :/

**Comment by MRizkBV** [comment]
I found a fix. The issue is not in Cisco AnyConnect client or even the OpenConnect server configuration. It is in the mobileconfig file generation. The mobileconfig file is bugged.

I created a mobileconfig file myself manually using the Apple Configurator 2 on Mac and it connected just fine using certificate (all certificates are on the streisand page). I even enabled On Demand and it connects just fine :)
